compiler/rustc_hir_typeck/src/fn_ctxt/adjust_fulfillment_errors.rs

```diff
@@ -48,20 +74,21 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         // Prefer generics that are local to the fn item, since these are likely
         // to be the cause of the unsatisfied predicate.
         let mut param_to_point_at = find_param_matching(&|param_ty| {
-            self.tcx.parent(generics.type_param(param_ty, self.tcx).def_id) == def_id
+            self.tcx.parent(generics.param_at(param_ty.idx(), self.tcx).def_id) == def_id
```

Comment:
this closure and the two below should be using `param_term` as the arg name instead of `param_ty`